extension/background.js

```diff
@@ -1,18 +1,54 @@
 var port = chrome.runtime.connectNative("default_opener");
 
+/**
+ * Call the native appliction with this url to open it with the default browser
+ */
+function callUrl(url) {
+    console.log(`Sending: link ${url} `);
+    port.postMessage({link: url});
+}
+
 port.onMessage.addListener((response) => {
-  console.log("Received: " + response);
+    console.log("Received: " + response);
 });
 
 port.onDisconnect.addListener(() => {
-  console.log("Port was disconnected. Error:", chrome.runtime.lastError);
+    console.log("Port was disconnected. Error:", chrome.runtime.lastError);
+    /* If we lost connection, simply reconnect. */
+    port = chrome.runtime.connectNative("default_opener");
```

Comment:
This looks like a good idea to recover but what would happen if there is an unrecoverable error? Does it try to reconnect in a loop forever?